### PR TITLE
fix: Adding close button to Dropdown list when it opens as a panel.

### PR DESCRIPTION
**Pull request by khmakoto**
## Current Behavior

`Dropdown` opens as a `Panel` on smaller screens, but this presents a problem for users of screen readers since they can't close the `Panel` as they are unable to access the backdrop to soft dismiss it.

## New Behavior

The `Panel` that opens when `Dropdown` is used on smaller screens now has a close button that can be accessed by screen reader users. Additionally, we are still keeping the focus by default going to the option list so there is no difference to how focus currently behaves when opening the `Dropdown`.

## Related Issue(s)

Fixes [ADO 14028](https://dev.azure.com/microsoftdesign/fluent-ui/_workitems/edit/14028).

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/95039a95033166fdd6c45478b1e9703302e17bb8'>95039a95033166fdd6c45478b1e9703302e17bb8</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react/src/components/Dropdown/Dropdown.base.tsx [hunk 1]
```diff
@@ -608,10 +608,10 @@ class DropdownInternal extends React.Component<IDropdownInternalProps, IDropdown
 
     return isSmall ? (
       <Panel
+        hasCloseButton
```
Comment (smhigley) [inline packages/react/src/components/Dropdown/Dropdown.base.tsx:611]: looks like it also needs `closeButtonAriaLabel={'close'}`

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/BaseButton-mount_base.html">915</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/BaseButton-mount.html">919</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Breadcrumb-mount_base.html">2676</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Breadcrumb-mount.html">2791</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Checkbox-mount_base.html">1492</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Checkbox-mount.html">1496</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/CheckboxBase-mount_base.html">1283</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/CheckboxBase-mount.html">1349</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/ChoiceGroup-mount_base.html">4722</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/ChoiceGroup-mount.html">4831</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/ComboBox-mount_base.html">1005</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/ComboBox-mount.html">956</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/CommandBar-mount_base.html">10495</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/CommandBar-mount.html">10666</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/ContextualMenu-mount_base.html">11781</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/ContextualMenu-mount.html">11705</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/DefaultButton-mount_base.html">1154</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/DefaultButton-mount.html">1129</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/DetailsRow-mount_base.html">3820</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/DetailsRow-mount.html">3926</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/DetailsRowFast-mount_base.html">3898</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/DetailsRowFast-mount.html">3807</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/DetailsRowNoStyles-mount_base.html">3657</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/DetailsRowNoStyles-mount.html">3729</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Dialog-mount_base.html">2841</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Dialog-mount.html">2828</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/DocumentCardTitle-mount_base.html">180</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/DocumentCardTitle-mount.html">172</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Dropdown-mount_base.html">3391</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Dropdown-mount.html">3364</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/FocusTrapZone-mount_base.html">1933</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/FocusTrapZone-mount.html">1858</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/FocusZone-mount_base.html">1870</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/FocusZone-mount.html">1855</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/IconButton-mount_base.html">1801</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/IconButton-mount.html">1774</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Label-mount_base.html">348</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Label-mount.html">350</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Layer-mount_base.html">3005</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Layer-mount.html">2972</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Link-mount_base.html">450</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Link-mount.html">498</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/MenuButton-mount_base.html">1549</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/MenuButton-mount.html">1523</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/MessageBar-mount_base.html">2197</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/MessageBar-mount.html">2120</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Nav-mount_base.html">3310</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Nav-mount.html">3378</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/OverflowSet-mount_base.html">1131</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/OverflowSet-mount.html">1087</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Panel-mount_base.html">2190</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Panel-mount.html">2220</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Persona-mount_base.html">1009</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Persona-mount.html">1021</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Pivot-mount_base.html">1477</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Pivot-mount.html">1475</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/PrimaryButton-mount_base.html">1338</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/PrimaryButton-mount.html">1273</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Rating-mount_base.html">7769</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Rating-mount.html">7880</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/SearchBox-mount_base.html">1315</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/SearchBox-mount.html">1344</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Shimmer-mount_base.html">2489</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Shimmer-mount.html">2445</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Slider-mount_base.html">1940</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Slider-mount.html">1949</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/SpinButton-mount_base.html">5046</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/SpinButton-mount.html">5093</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Spinner-mount_base.html">464</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Spinner-mount.html">449</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/SplitButton-mount_base.html">3197</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/SplitButton-mount.html">3204</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Stack-mount_base.html">518</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Stack-mount.html">529</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/StackWithIntrinsicChildren-mount_base.html">2294</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/StackWithIntrinsicChildren-mount.html">2320</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/StackWithTextChildren-mount_base.html">5341</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/StackWithTextChildren-mount.html">5575</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/SwatchColorPicker-mount_base.html">11849</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/SwatchColorPicker-mount.html">11722</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/TagPicker-mount_base.html">2775</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/TagPicker-mount.html">2681</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/TeachingBubble-mount_base.html">92821</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/TeachingBubble-mount.html">92067</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Text-mount_base.html">438</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Text-mount.html">418</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/TextField-mount_base.html">1395</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/TextField-mount.html">1438</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/ThemeProvider-mount_base.html">1187</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/ThemeProvider-mount.html">1187</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/ThemeProvider-virtual-rerender_base.html">626</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/ThemeProvider-virtual-rerender.html">629</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1933</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1977</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Toggle-mount_base.html">848</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/Toggle-mount.html">808</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/buttonNative-mount_base.html">130</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23524/perf-test/buttonNative-mount.html">130</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/components/Dropdown/Dropdown.base.tsx [hunk 2]
```diff
@@ -664,6 +664,7 @@ class DropdownInternal extends React.Component<IDropdownInternalProps, IDropdown
           aria-label={ariaLabel}
           aria-labelledby={label && !ariaLabel ? this._labelId : undefined}
           aria-multiselectable={multiSelect}
+          shouldFocusOnMount
```
Comment (smhigley) [inline packages/react/src/components/Dropdown/Dropdown.base.tsx:667]: it still seems to be focusing on the close button on open for me. I don't think that's the worst thing in the world, but I think the panel focus is happening after the focuszone focus and winning.

Comment (khmakoto) [reply packages/react/src/components/Dropdown/Dropdown.base.tsx]: I've changed my approach, let me know if it still doesn't work.

**Review by smhigley — APPROVED** [review]

**Pull request merged**